src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,127 @@
+
+        while (!request.equals("bye")) {
+            Parser parser = new Parser(sc.nextLine());
+            request = parser.getRequest();
+            String args = parser.getArgs();
+
+            if (request.equals("bye")) {
+                ui.showBye();
+                break;
+            } else if (request.equals("list")) {
+                ui.printList(tasks);
+            } else if (request.equals("done")) {
+                try {
+                    int taskNo = Integer.parseInt(args);
+                    ui.printMarked(tasks.markDone(taskNo));
+                    storage.save(tasks);
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                } catch (NumberFormatException ex) {
+                    ui.printFormatted("Please enter integer values..");
+                }
+            } else if (request.equals("delete")) {
+                try {
+                    int taskNo = Integer.parseInt(args);
+                    ui.printRemoved(tasks, tasks.removeTask(taskNo));
+                    storage.save(tasks);
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                } catch (NumberFormatException ex) {
+                    ui.printFormatted("Please enter integer values..");
+                }
+            } else if (request.equals("todo")) {
+                try {
+                    Task task = Task.createTask(args, request, "", "");
+                    ui.printAdded(tasks, tasks.addTask(task));
+                    storage.save(tasks);
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                }
+            } else if (request.equals("deadline")) {
+                try {
+                    String[] deadStr = parser.getFormattedCommand();
+                    Task task = Task.createTask(deadStr[0], request, deadStr[1], deadStr[2]);
+                    ui.printAdded(tasks, tasks.addTask(task));
+                    storage.save(tasks);
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                }
+            } else if (request.equals("event")) {
+                try {
+                    String[] eventStr = parser.getFormattedCommand();
+                    Task task = Task.createTask(eventStr[0], request, eventStr[1], eventStr[2]);
+                    ui.printAdded(tasks, tasks.addTask(task));
+                    storage.save(tasks);
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                }
+            } else if (request.equals("find")) {
+                ui.printFound(tasks.findTask(args));
+            } else {
+                try {
+                    throwDK();
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                }
+            }
+        }
```

Comment:
Perhaps try to avoid deep nesting and code duplication here by instead using an outer `try catch` block that catches the same `DukeException` that each `if` clause throws?